ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,220 @@
+# 47. GitOps Onboarding Redesign
+
+Date: 2025-07-09
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, Konflux offers two distinct onboarding paths that create friction and complexity for developers:
+
+### UI Onboarding
+- Very easy to use with good validation and feedback
+- Handles secrets, build config, environments through web interface
+- Provides immediate feedback but doesn't align with GitOps principles
+
+### GitOps Onboarding
+- Requires manually authoring YAML across multiple files
+- Validation happens during GitLab CI runs, creating tension between Konflux and GitLab CI workflows
+- Secrets must still be created via UI, breaking the GitOps flow
+- Onboarding is slow and often requires multiple failed PRs to converge due to trial-and-error approach
+
+### Current Pain Points
+1. **Fragmented experience**: Developers must context-switch between UI and GitOps approaches
+2. **Validation delays**: Schema validation only occurs during CI runs, not locally
+3. **Error-prone process**: Manual YAML authoring leads to frequent failures
+4. **Secrets inconsistency**: Secrets cannot be managed declaratively through Git
+5. **Monorepo complexity**: Current GitOps monorepo for tenant configuration creates CODEOWNERS management overhead
+6. **UI improvement limitations**: Despite continuous improvements to the UI, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits, leaving them unable to benefit from UI enhancements
+
+## Decision
+
+Redesign the onboarding experience to converge UI and GitOps into a single coherent, Git-centric flow through the following architectural changes:
+
+### Alternative Approaches Considered
+
+**Improving the existing UI** was considered but rejected because:
+- Despite continuous UI improvements, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits
+- UI enhancements don't address the fundamental issue of maintaining two separate onboarding flows
+- Users who prefer GitOps principles cannot benefit from UI improvements, creating a permanent split in the user experience
+
+**Chosen approach** replaces both current flows with a unified Git-centric solution that preserves GitOps benefits while providing the usability improvements users expect.
+
+### Core Shift
+- **Replace UI configuration** with a VS Code plugin for onboarding and editing
+- **Use local schema validation**, linting, and previews to reduce onboarding errors
+- **Use Git as the single source of truth** for applications, components and secrets
```

Comment:
What about integration test scenarios and releaseplans?

Perhaps just say "tenant resources".

Reply:
It probably also makes sense to move that to the gitops based approach.